packages/wallets/src/components/WalletImport.tsx

```diff
@@ -43,11 +56,60 @@ export default function WalletImport() {
     },
   });
 
-  const { fields } = useFieldArray({
+  const { fields, replace } = useFieldArray({
     control: methods.control,
     name: 'mnemonic',
   });
 
+  const submitMnemonicPaste = (mnemonicList: string) => {
+    closeMnemonicPaste();
+    let mList = mnemonicList.match(/\b(\w+)\b/g);
+
+    if (mList == null) {
+      openDialog(
+        <AlertDialog>
+          <Trans>
+            Your pasted list does not include 24 valid mnemonic words.
+          </Trans>
+        </AlertDialog>
+      );
+      return;
+    }
+
+    const intersection = mList.filter(element => options.includes(element));
+
+    if (intersection.length != 24) {
+      openDialog(
+        <AlertDialog>
+          <Trans>
+            Your pasted list does not include 24 valid mnemonic words.
+          </Trans>
+        </AlertDialog>
+      );
+    } else {
+      const mnemonic = intersection.map((word) => ({ word }));
+      replace(mnemonic);
+    }
+
+    // above abandon abstract absurd account acid alert alley blast bless blind blanket boost book busy burden burger census chair chalk dinner direct dirt discover
```

Comment:
Remove?

Reply:
ha, i'd taken it out once before but it snuck back in